### PR TITLE
Upped php version to 7.2.0, some initial basic code

### PR DESCRIPTION
**Pull request by ghost**
check for any errors with new php version

### CODE REVIEW

File: public/dnd.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        /**
+         * Created by PhpStorm.
+         * User: Luka_offline
+         * Date: 15/04/2018
+         * Time: 20:13
+         */
+
+        $dice_data = $_GET["roll"];
+
+        $dice_arr = preg_split("/d|\+/", $dice_data);
```
Comment (judahnator) [inline public/dnd.php]: Most of this looks good, except for this bit here.

Plus signs are really hit and miss in URLs, they are often interpreted as spaces.

What if we split out the GET variables instead?
* n => number of dice
* d => sides per dice
* o => offset

**Comment by judahnator** [comment]
You can ignore the StyleCI thing, I will deal with that later

File: public/dnd.php (hunk 1 above)
Comment (ghost) [reply public/dnd.php]: I need to check if nightbot can do that, for now, my next commit assumes + works well :P

**Comment by judahnator** [comment]
I have what you wrote integrated with the framework.

Tell me what you think @LukaPrebil.

649e9927c5d8b41ceb90317f4abb3e783196cc44

**Comment by ghost** [comment]
Looks good to me. Where will this live in the end? @judahnator

**Comment by judahnator** [comment]
This has been deployed to production.
You can see it live at https://beta.uoltt.org/api/dnd

Here is an example query: https://beta.uoltt.org/api/dnd?roll=4d20+2

**Pull request merged**